src/movies.js

```diff
@@ -1,25 +1,140 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const mapMovies = moviesArray.map(movie => movie.director);
+    return mapMovies;
+}
+//console.log(getAllDirectors(movies));
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    if (!moviesArray.length) return 0;
+    const spielbergMovies = moviesArray.filter(movie => movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'));
+    //console.log(spielbergMovies);
+    return spielbergMovies.length;
+}
+//console.log(howManyMovies(movies))
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (!moviesArray.length) return 0;
```

Comment:
The reduce function won't execute if the moviesArray is empty. 
The code block `if (!moviesArray.length) return 0;` is not needed and the function will work fine without it, it will return 0 as default value for empty array.